src/OpenTelemetry.Api/Internal/ActivityHelperExtensions.cs

```diff
@@ -265,7 +270,7 @@ private static class ActivityTagsEnumeratorFactory<TState>
             private static readonly DictionaryEnumerator<string, object, TState>.ForEachDelegate ForEachTagValueCallbackRef = ForEachTagValueCallback;
 
             [MethodImpl(MethodImplOptions.AggressiveInlining)]
-            public static void Enumerate(Activity activity, ref TState state)
+            public static void Enumerate(Activity activity, ref TState state, int? maxTags)
```

Comment:
If we don't go with the overload approach, then maybe assign `maxTags` a default value of `null`.